x-pack/plugins/security/server/authentication/api_key.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export interface CreateApiKeyOptions {
+  callCluster: any;
+  isSecurityFeatureDisabled: () => boolean;
+  body: {
+    name: string;
+    role_descriptors: Record<string, any>;
+    expiration?: string;
+  };
+}
+
+export interface CreateApiKeyResult {
+  id: string;
+  name: string;
+  expiration?: string;
+  api_key: string;
+}
+
+export async function createApiKey({
+  callCluster,
+  body,
+  isSecurityFeatureDisabled,
+}: CreateApiKeyOptions): Promise<CreateApiKeyResult | null> {
+  if (isSecurityFeatureDisabled()) {
+    return null;
+  }
+  return await callCluster('shield.createApiKey', { body });
```

Comment:
__nit:__ it'd be great if we can at least debug-log here whenever new API key is requested. Later, it feels, we'll need to issue an audit log event here as well.... (ping @kobelb).

All in all it seems we'll end up with a `APIKeys.create/invalidate/retrieve/*` class here that can accept logger/auditLogger/clusterClient/isSecurityFeatureDisabled at the initialization (constructor) stage :) Having said that I'm totally fine to keep it as is right now if you're not up to this kind of generalization at this stage.

Reply:
For the time being, I think we can ignore the audit logging here... We only *have* to write entries to the Kibana audit log when we're performing authentication/authorization ourselves in Kibana and can't defer to the Elasticsearch audit log. At least, that's been the current thinking, which is entirely up for debate once we start to focus more on audit logging.

Reply:
> We only have to write entries to the Kibana audit log when we're performing authentication/authorization ourselves in Kibana and can't defer to the Elasticsearch audit log.

Right, good point, thanks!

Reply:
Should the comment stay for now or be removed?